zcash_primitives/src/transaction/builder.rs

```diff
@@ -134,21 +175,47 @@ impl Progress {
 }
 
 /// Generates a [`Transaction`] from its inputs and outputs.
-pub struct Builder<'a, P, R> {
+pub struct Builder<'a, P, R, O: MaybeOrchard = WithoutOrchard> {
     params: P,
     rng: R,
     target_height: BlockHeight,
     expiry_height: BlockHeight,
     transparent_builder: TransparentBuilder,
     sapling_builder: SaplingBuilder<P>,
+    orchard_builder: O,
+    // TODO: These should be calculated by the orchard builder and not cached here
+    // This requires publicizing methods or fields of the orchard builder
+    num_orchard_spends: usize,
+    num_orchard_outputs: usize,
+    orchard_saks: Vec<orchard::keys::SpendAuthorizingKey>,
```

Comment:
Can you please open a PR against the `orchard` repository that proposes the required changes? I want to make sure that this is not misunderstanding the Orchard builder API; for example, the Orchard builder is used within the `zcashd` transaction builder unmodified.

Reply:
Orchard PR here: https://github.com/zcash/orchard/pull/365. There are more changes needed, but I may as well get simple changes out-of-the-way and not risk making a too-big PR.